code/__HELPERS/game.dm

```diff
@@ -535,6 +535,25 @@ block( \
 	announcer.announce("ARRIVAL", character.real_name, displayed_rank, list()) //make the list empty to make it announce it in common
 	//WS end
 
+/proc/NotifyFaction(mob/living/carbon/human/character, obj/structure/overmap/ship/simulated/ship)
```

Comment:
This whole proc looks good, but 
1. It needs some DMDocs.
Essentially just before the proc you would put:
```suggestion
/**
 * short description of the proc
 *
 * Arguments:
 * * mob/living/carbon/human/character - short description of the arg
 * * arg2 - etc
 */
 /proc/NotifyFaction........
```
2. You have return in every single case, along with the else.
Why not take the return out of everyone, remove the else and just have a return outside of the switch statement?

Reply:
Should be good to go